### PR TITLE
feat(markdown-slate): push empty paragraph onto slate dom

### PR DESCRIPTION
**Pull request by irmerk**
# No Issue

### Changes
- Create an empty Slate DOM paragraph block and to tack onto any transform from CiceroMark

### Flags
- Still will require Slate work to handle runtime editing and adding a template to the end of a contract

### Related Issues
- N/A

### CODE REVIEW

File: packages/markdown-slate/src/SlateTransformer.js [hunk 1]
```diff
@@ -51,6 +51,20 @@ class SlateTransformer {
...
+            },
+            nodes: [
+                {
+                    object: 'text',
+                    text: '',
+                    marks: []
+                }
+            ]
+        };
+        result.document.nodes.push(paragraphSpaceNodeJSON);
```
Comment (dselman) [inline packages/markdown-slate/src/SlateTransformer.js]: If you only added this when the last block was a Clause then you wouldn't need to change all the tests... ;-)

File: packages/markdown-slate/src/slateToCiceroMarkDom.js [hunk 2]
```diff
@@ -32,6 +32,19 @@ function slateToCiceroMarkDom(document) {
     // convert the value to a plain object
     const json = JSON.parse(JSON.stringify(document));
     _recursive(result, json.nodes);
+    const isLastNodePara = result.nodes[result.nodes.length - 1]
```
Comment (jeromesimeon) [inline packages/markdown-slate/src/slateToCiceroMarkDom.js]: This should be made a bit more defensive so it does not fail when the array is empty.

Comment (jeromesimeon) [reply packages/markdown-slate/src/slateToCiceroMarkDom.js]: Also this should only apply when the last paragraph is preceded by a clause node.

Comment (jeromesimeon) [reply packages/markdown-slate/src/slateToCiceroMarkDom.js]: Minor code style: `result.nodes[result.nodes.length - 1]` can be put in a variable, somewhat more concise.

File: packages/markdown-slate/src/slateToCiceroMarkDom.js [hunk 3]
```diff
@@ -32,6 +32,19 @@ function slateToCiceroMarkDom(document) {
...
+        .$class === 'org.accordproject.commonmark.Paragraph';
+    const isPragraphEmpty = result.nodes[result.nodes.length - 1].nodes
```
Comment (jeromesimeon) [inline packages/markdown-slate/src/slateToCiceroMarkDom.js]: Minor code style: `isParagraphEmpty` ?

File: packages/markdown-slate/src/slateToCiceroMarkDom.js [hunk 4]
```diff
@@ -32,6 +32,19 @@ function slateToCiceroMarkDom(document) {
...
+        ? result.nodes[result.nodes.length - 1].nodes[0].text === ''
+        : null;
+    if (isLastNodePara && isPragraphEmpty) {
+        const removeByIndex = (array, index) => array.filter((_, i) => i !== index);
```
Comment (jeromesimeon) [inline packages/markdown-slate/src/slateToCiceroMarkDom.js]: Maybe would look nicer with: `result.nodes.splice(0, result.nodes.length-1);` ?

File: packages/markdown-slate/src/slateToCiceroMarkDom.js [hunk 5]
```diff
@@ -17,6 +17,33 @@
 const NS = 'org.accordproject.commonmark';
 const NS_CICERO = 'org.accordproject.ciceromark';
 
+const lastNode = (input) => input.nodes[input.nodes.length - 1];
+
+const isSecondLastClause = (input) => input.nodes[input.nodes.length - 2]
```
Comment (jeromesimeon) [inline packages/markdown-slate/src/slateToCiceroMarkDom.js]: This call will still raise an `undefined` error if there are less than two nodes in the input.
I think those would be best written as `input.nodes[input.nodes.length - 1] && input.nodes[input.nodes.length - 1].$class` etc.

**Pull request merged**